File: 2020/179.py
"""
179
"""
from functools import cmp_to_key
from typing import List
class Solution:
    def largestNumber(self, nums: List[int]) -> str:
        A = [str(x) for x in nums]
        A.sort(key=cmp_to_key(lambda x, y: -1 if x+y>y+x else y+x>x+y))
        return ''.join(A).lstrip('0') or '0'

print(Solution().largestNumber([10,2]))

File: 2020/179.java
import java.util.*;
import java.util.stream.Collectors;

class Solution {
    public String largestNumber(int[] nums) {
        List<String> arr = Arrays.stream(nums).mapToObj(i -> ((Integer) i).toString()).collect(Collectors.toList());
        arr.sort((String s1, String s2) -> (s2+s1).compareTo(s1+s2));
        // regex: find one or more leading zeros if not followed by the end of string
        return String.join("", arr).replaceAll("^0+(?!$)", "");
    }
}